src/clj/game/core/rules.clj

```diff
@@ -228,33 +228,31 @@
   ([state side type n] (damage state side (make-eid state) type n nil))
   ([state side type n args] (damage state side (make-eid state) type n args))
   ([state side eid type n {:keys [unpreventable unboostable card] :as args}]
-   (swap! state update-in [:damage :damage-bonus] dissoc type)
-   (swap! state update-in [:damage :damage-prevent] dissoc type)
-   ;; alert listeners that damage is about to be calculated.
+   (swap! state update-in [:damage :damage-bonus] dissoc type)   ; cleans up any bonus from previous damage
+   (swap! state update-in [:damage :damage-prevent] dissoc type) ; cleans up any prevention from previous damage
+    ;; alert listeners that damage is about to be calculated.
    (trigger-event state side :pre-damage type card n)
-   (let [n (damage-count state side type n args)]
-     (let [prevent (get-in @state [:prevent :damage type])]
-       (if (and (not unpreventable) prevent (pos? (count prevent)))
-         ;; runner can prevent the damage.
-         (do (system-msg state :runner "has the option to avoid damage")
-             (show-wait-prompt state :corp "Runner to prevent damage" {:priority 10})
-             (swap! state assoc-in [:prevent :current] type)
-             (show-prompt
-               state :runner nil (str "Prevent any of the " n " " (name type) " damage?") ["Done"]
-               (fn [_]
-                 (let [prevent (get-in @state [:damage :damage-prevent type])]
-                   (when prevent
-                     (trigger-event state side :prevented-damage type prevent))
-                   (system-msg state :runner
-                               (if prevent
-                                 (str "prevents " (if (= prevent Integer/MAX_VALUE) "all" prevent)
-                                      " " (name type) " damage")
-                                 "will not prevent damage"))
-                   (clear-wait-prompt state :corp)
-                   (resolve-damage state side eid type (max 0 (- n (or prevent 0))) args)))
-               {:priority 10}))
-         (resolve-damage state side eid type n args))))))
-
+   (let [n (damage-count state side type n args)
+         prevent (get-in @state [:prevent :damage type])]
+     (if (and (not unpreventable) (cards-can-prevent? state side prevent :damage type))
+       ;; runner can prevent the damage.
+       (do (system-msg state :runner "has the option to avoid damage")
+           (show-wait-prompt state :corp "Runner to prevent damage" {:priority 10})
+           (swap! state assoc-in [:prevent :current] type)
+           (show-prompt state :runner nil (str "Prevent any of the " n " " (name type) " damage?") ["Done"]
+                        (fn [_]
+             (let [prevent (get-in @state [:damage :damage-prevent type])]
```

Comment:
Indentation here is a little misleading.

Reply:
will fix!